crates/codegen/language/definition/src/model/identifier.rs

```diff
@@ -0,0 +1,45 @@
+use serde::Serialize;
+use std::ops::Deref;
+
+/// A wrapper type to make sure the DSL token is written as an identifier instead of a string literal.
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct Identifier {
```

Comment:
I'm going to risk a dumb question - could we use proc_macro2/syn::Ident here directly, since we only seem to use it with the macro and we almost always want it with a span?

Reply:
We cannot use it instead of `Identifier`, as we cannot implement other traits for this external type.
We cannot wrap it inside `Identifier` either, as we would have to special case it from `internal_macros` in order for it to not be wrapped unnecissarily in `Spanned<>` again.
Also, it is not `serde` serializable, at least not without workarounds, which we need for the entire model.

Reply:
Got it, thanks! Maybe it'd be worthwhile to add a comment why we're not using `proc_macro2::Ident` directly?